client/src/main/java/org/eclipse/hono/client/HonoClient.java

```diff
@@ -68,6 +68,18 @@
      */
     Future<HonoClient> connect();
 
+    /**
+     * Disconnects from the Hono server.
```

Comment:
what is the purpose of *disconnecting*? We should provide some information to the client as to what might be possible reasons why a client would/should disconnect. It would also help to know if the client can be re-connected again once it has been disconnected.

Reply:
thanks! `disconnecting` is a flag that indicates whether or not the client is in the middle of a disconnect operation. my understanding is that the `disconnect` method is called directly on a client instance to explicitly disconnect from the server (for reasons described in #500). 
i also think that a `HonoClient#disconnect()` call should be followed by a subsequent `HonoClient#connect()` if the client wants to connect back to the server after disconnecting. is that a valid assumption?

Reply:
please document this in the JavaDoc comment so that users of the API have a chance to learn about this as well ...

Reply:
ok, thanks